pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -1032,12 +1032,20 @@ func containerMemoryFromBlock(blocks []memorymanagerstate.Block) []*podresources
 	return containerMemories
 }
 
-func (cm *containerManagerImpl) PrepareResources(pod *v1.Pod, container *v1.Container) (*dra.ContainerInfo, error) {
-	return cm.draManager.PrepareResources(pod, container)
+func (cm *containerManagerImpl) PrepareResources(pod *v1.Pod, container *v1.Container) error {
+	if cm.draManager != nil {
+		return cm.draManager.PrepareResources(pod, container)
+	}
+
+	return nil
```

Comment:
cm.draManager can't be nil unless `DynamicResourceAllocation` feature is enabled ,so this check is the same as  `if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DynamicResourceAllocation) {`.
I decided to check for nil to be consistent with the code in this file, e.g. with https://github.com/kubernetes/kubernetes/blob/release-1.26/pkg/kubelet/cm/container_manager_linux.go#L714

Reply:
Checked if `DynamicResourceAllocation` feature is enabled instead. This is better as we'll hopefully need to remove this check when DRA is GA. It will be easier to find it.